source/common/router/header_parser.cc

```diff
@@ -161,6 +161,14 @@ void HeaderParser::evaluateHeaders(Http::HeaderMap& headers,
     } else {
       value = entry.original_value_;
     }
+
+    // Skip any invalid header value.
+    // The validity of the header key has already been checked as it is generated by
+    // Http::LowerCaseString.
+    if (!Envoy::Http::validHeaderString(value)) {
+      continue;
+    }
```

Comment:
Sorry in the real code where is this check done? By the codec? If this is purely for fuzzing can you potentially just guard these blocks with a FUZZ ifdef? Seems not great to add this extra runtime check where it is not needed.

Reply:
The validation in codec for example, [http2](https://github.com/envoyproxy/envoy/blob/6712ff66ab3c93904e08aae7f139d4ecdf89d225/source/common/http/http2/codec_impl.cc#L249) and [http1](https://github.com/envoyproxy/envoy/blob/6712ff66ab3c93904e08aae7f139d4ecdf89d225/source/common/http/http1/codec_impl.cc#L449). and I think UHV is the long-term solution to unify those validations. 

I agree with undesired runtime cost (as I also mentioned it in the PR discussion). Let me take another look to see if this can be fixed by rejecting fuzz config.  Thanks!